scripts/findAndMatch.js

```diff
@@ -0,0 +1,319 @@
+    outcome.children[i].value = randomNames[i];
+  }
+  event.target.innerText = event.target.value;
+  event.target.style.cssText =
+    "background-color: #000000; color: #ffffff; transition: 0.3s ease-in-out;";
+}
+
+function hideName() {
+  event.target.innerText = "Hidden";
+  event.target.style.cssText = "background-color: #ffffff; color: #000000;";
+}
+
+function guessName() {
+  if (event.target.innerText == key) {
+    var count = 0;
+    count++;
+    var x = randomNames.indexOf(event.target.innerText);
+    outcome.removeChild(outcome.childNodes[x + 1]);
+    randomNames.splice(x, 1);
+    outcome.appendChild(p).innerHTML = "Correct!";
+    p.style.cssText = "color: darkseagreen;";
+    findArtist();
+    if (key == undefined) {
+      msg1.innerHTML =
+        "CONGRATULATIONS, you won the 1st round!!!" +
+        "<br><br>" +
+        'You can play again by pressing "Start" or continue to part II, below!' +
+        "<br>" +
+        "Remeber that part II is based on dragging an artist's portrait to his / her corresponding instrument.";
+      msg1.style.cssText = "color: darkseagreen; font-size: 24px;";
+      msg2.innerHTML = "";
+      startBtn.disabled = false;
+      document.body.removeChild(outcome);
+      document.body.removeChild(instructions);
+      msg3.innerHTML = "";
+      artists.style.cssText = "visibility: visible;";
+      instruments.style.cssText = "visibility: visible;";
+      stopCounter();
+      part2();
+    }
+  } else {
+    outcome.appendChild(p).innerHTML = "Incorrect!";
+    p.style.cssText = "color: indianred;";
+  }
+}
+
+/*********************************************************************************
+PART II
+*********************************************************************************/
+var artists = document.getElementById("artists");
+var instruments = document.getElementById("instruments");
+
+var guitarists = Array.from(document.getElementsByClassName("guitarist"));
+var vocalists = Array.from(document.getElementsByClassName("vocalist"));
+var bassists = Array.from(document.getElementsByClassName("bassist"));
+var drummers = Array.from(document.getElementsByClassName("drummer"));
+
+function part2() {
+  artists.style.cssText = "visibility: visible";
+  instruments.style.cssText = "visibility: visible";
+  cheatCounter();
+}
+
+//Event listeners and draggable assignments
+for (var i = 1; i < artists.children.length; i++) {
+  artists.children[i].draggable = true;
+}
+
+for (var i = 1; i < artists.children.length; i++) {
+  artists.children[i].addEventListener("dragstart", startDrag);
+}
+
+for (var i = 1; i < instruments.children.length; i++) {
+  instruments.children[i].addEventListener("dragover", dragOver);
+}
+
+for (var i = 1; i < instruments.children.length; i++) {
+  instruments.children[i].addEventListener("drop", dropDrag);
+}
+
+//Drag and drop functionality
+var holderItem;
+
+function startDrag() {
+  holderItem = event.target;

```

Comment:
In all the functions that are added as event listener you have to declare the event parameter. The browser calls the function you provide when an event happens with the event. So in the Chrome's code it's something like this:
```
var event;
if (mouseOver) {
       startDrag(event)
}
```
That way it becomes available in your function. In this case it because some browsers are clever, but you can't rely on that. (Maybe that's why it's not working in other browsers)